### ISSUE TITLE
PestPHP fails when PHPUnit is running in test isolation

### ISSUE THREAD

**Issue by mglaman** [pestphp/pest#665]
tl;dr I'm not sure of a path forward beyond just a documentation piece that says PestPHP cannot execute if PHPUnit is running tests in separate processes.

----

I was working on trying to test Drupal with PestPHP using the following line to use an existing base test from Drupal.

```
uses(
  \Drupal\KernelTests\KernelTestBase::class,
)->in('Kernel');
```

When I executed the test, I got an error about the evaluated code from PHPUnit.

```
  PHP Parse error:  syntax error, unexpected identifier "d" in Standard input code on line 34
```

I did some debugging and PHPUnit was running the test in isolation and provided the following in its template (full source here: https://gist.github.com/mglaman/8ffe67a117f2aa4f41839420d2e7abb9)

```
  if (!class_exists('P\Tests\Kernel\ExampleTest')) {
    require_once '/Users/mglaman/Sites/pest-drupal/vendor/pestphp/pest/src/Factories/TestCaseFactory.php(223) : eval()'d code';
    }

    $result = new PHPUnit\Framework\TestResult;
```

The `class_exists` is failing because the test is loaded in a subprocess. In `\Pest\Factories\TestCaseFactory::makeClassFromFilename` it uses `eval` to create the class at runtime in memory, so it exists in the original process.

I'm not sure of a path forward beyond just a documentation piece that says PestPHP cannot execute if PHPUnit is running tests in separate processes.

**Comment by Frank-Landry** [comment]
@mglaman This might help? https://pestphp.com/docs/plugins/parallel

**Comment by dansysanalyst** [comment]
Hi @mglaman, according to Nuno in  https://github.com/pestphp/pest/issues/618 Pest does not support process isolation.

**Comment by mglaman** [comment]
Thanks @dansysanalyst, I'll close this then since I do not right now have capacity to add support.
